main.cpp

```diff
@@ -19,13 +18,31 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+	CastCallBack(Rewriter& rewriter) : rewriter_(rewriter) { }
 
-    void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+	virtual void run(const MatchFinder::MatchResult& Result) {
+	const auto *CastExpr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+	auto &SM = *Result.SourceManager;
+
+	auto DestTypeString = Lexer::getSourceText(CharSourceRange::getTokenRange(

```

Comment:
Please describe what the `getSourceText()` function does.

Reply:
getSourceText() returns a string for the source that the range encompasses.

 In a  code the range is:
  getTokenRange(cast_expr->getLParenLoc().getLocWithOffset(1), cast_expr->getRParenLoc().getLocWithOffset(-1))